src/python_testing/TC_SUTestBase.py

```diff
@@ -0,0 +1,480 @@
+                                      event_report: Clusters.OtaSoftwareUpdateRequestor.Events.StateTransition,
+                                      previous_state,
+                                      new_state,
+                                      target_version: Optional[int] = None,
+                                      reason: Optional[int] = None):
+        """Verify the values of the StateTransitionEvent from the EventHandler given the provided arguments.
+
+        Args:
+            event_report (Clusters.OtaSoftwareUpdateRequestor.Events.StateTransition): StateTransition Event report to verify.
+            previous_state (UpdateStateEnum:int): Int or UpdateStateEnum value for the previous state.
+            new_state (UpdateStateEnum:int): Int or UpdateStateEnum value for the new or current state.
+            target_version (Optional[int], optional): Software version to verify if not provided ignore this check.. Defaults to None.
+            reason (Optional[int], optional): UpdateStateEnum reason of the event, if not provided ignore. Defaults to None.
+        """
+        logger.info(f"Verifying the event {event_report}")
+        asserts.assert_equal(event_report.previousState, previous_state, f"Previous state was not {previous_state}")
+        asserts.assert_equal(event_report.newState,  new_state, f"New state is not {new_state}")
+        if target_version is not None:
+            asserts.assert_equal(event_report.targetSoftwareVersion,  target_version, f"Target version is not {target_version}")
+        if reason is not None:
+            asserts.assert_equal(event_report.reason,  reason, f"Reason is not {reason}")
+
+    def create_acl_entry(self,
+                         dev_ctrl: ChipDeviceCtrl.ChipDeviceController,
+                         provider_node_id: int,
+                         requestor_node_id: Optional[int] = None,
+                         acl_entries: Optional[list[Clusters.AccessControl.Structs.AccessControlEntryStruct]] = None,
+                         ):
+        """Create ACL entries to allow OTA requestors to access the provider.
+
+        Args:
+            dev_ctrl: Device controller for sending commands
+            provider_node_id: Node ID of the OTA provider
+            requestor_node_id: Optional specific requestor node ID for targeted access
+            acl_entries: Optional[list[Clusters.AccessControl.Structs.AccessControlEntryStruct]]. ACL list to write ino the requestor.
+
+        Returns:
+            Result of the ACL write operation
+        """
+        # Standard ACL entry for OTA Provider cluster
+        admin_node_id = dev_ctrl.nodeId if hasattr(dev_ctrl, 'nodeId') else self.DEFAULT_ADMIN_NODE_ID
+        requestor_subjects = [requestor_node_id] if requestor_node_id else NullValue
+
+        if acl_entries is None:
+            # If there are not ACL entries using proper struct constructors create the default.
+            acl_entries = [
```

Comment:
nit: probably it is worth to put it as a constant in the top of the class: `DEFAULT_ACL_ENTRIES`